lib/ansible/modules/cloud/amazon/cloudfront_facts.py

```diff
@@ -553,27 +542,41 @@ def set_facts_for_distribution_id_and_alias(details, facts, distribution_id, ali
 def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(dict(
-        distribution_id=dict(required=False, type='str'),
-        invalidation_id=dict(required=False, type='str'),
-        origin_access_identity_id=dict(required=False, type='str'),
-        domain_name_alias=dict(required=False, type='str'),
-        all_lists=dict(required=False, default=False, type='bool'),
-        distribution=dict(required=False, default=False, type='bool'),
-        distribution_config=dict(required=False, default=False, type='bool'),
-        origin_access_identity=dict(required=False, default=False, type='bool'),
-        origin_access_identity_config=dict(required=False, default=False, type='bool'),
-        invalidation=dict(required=False, default=False, type='bool'),
-        streaming_distribution=dict(required=False, default=False, type='bool'),
-        streaming_distribution_config=dict(required=False, default=False, type='bool'),
-        list_origin_access_identities=dict(required=False, default=False, type='bool'),
-        list_distributions=dict(required=False, default=False, type='bool'),
-        list_distributions_by_web_acl_id=dict(required=False, default=False, type='bool'),
-        list_invalidations=dict(required=False, default=False, type='bool'),
-        list_streaming_distributions=dict(required=False, default=False, type='bool'),
-        summary=dict(required=False, default=False, type='bool')
+        distribution_id=dict(type='str'),
+        invalidation_id=dict(type='str'),
+        origin_access_identity_id=dict(type='str'),
+        domain_name_alias=dict(type='str'),
+        all_lists=dict(type='bool', default=False),
+        distribution=dict(type='bool', default=False),
+        distribution_config=dict(type='bool', default=False),
+        origin_access_identity=dict(type='bool', default=False),
+        origin_access_identity_config=dict(type='bool', default=False),
+        invalidation=dict(type='bool', default=False),
+        streaming_distribution=dict(type='bool', default=False),
+        streaming_distribution_config=dict(type='bool', default=False),
+        list_origin_access_identities=dict(type='bool', default=False),
+        list_distributions=dict(type='bool', default=False),
+        list_distributions_by_web_acl_id=dict(type='bool', default=False),
+        list_invalidations=dict(type='bool', default=False),
+        list_streaming_distributions=dict(type='bool', default=False),
+        summary=dict(type='bool', default=False),
     ))
 
-    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False,
+        required_by=dict(
+            distribution=('distribution_id', 'domain_name_alias'),
```

Comment:
Isn't at least one of `distribution_id` and `domain_name_alias` required for `distribution`? And not necessarily both? (At least from the check below which used to be in line 617.)

Reply:
Ypu, fixed.